concent_api/core/tests/test_core_views.py

```diff
@@ -286,6 +286,40 @@ def test_send_reject_message_save_as_receive_out_of_band_status(self):
         self.assertEqual(reject_response.status_code, 202)
         self.assertEqual(Message.objects.last().type, message.RejectReportComputedTask.TYPE)
 
+    def test_send_should_return_http_400_if_task_to_compute_deadline_is_wrong(self):
+        compute_task_def = message.ComputeTaskDef()
+        compute_task_def['task_id'] = '8'
+
+        for deadline in [-11, 'a11b', int(dateutil.parser.parse("2017-11-17 11:00:00").timestamp())]:
+            compute_task_def['deadline'] = deadline
+            task_to_compute = message.TaskToCompute(
+                timestamp = int(datetime.datetime.now().timestamp()),
+                compute_task_def = self.compute_task_def,
+            )
+
+            serialized_task_to_compute      = dump(task_to_compute,             REQUESTOR_PRIVATE_KEY,   PROVIDER_PUBLIC_KEY)
+            deserialized_task_to_compute    = load(serialized_task_to_compute,  PROVIDER_PRIVATE_KEY,  REQUESTOR_PUBLIC_KEY, check_time = False)
+
+            ack_report_computed_task = message.AckReportComputedTask(
+                timestamp = int(datetime.datetime.now().timestamp()),
```

Comment:
Don't use `datetime.now()` outside of `freeze_time()`. If you need it here, make the `with freeze_time()` block a little bigger. It should cover everything that depends on current time.

Reply:
:+1: